promotion/drain.go

```diff
@@ -385,37 +384,57 @@ func (service *Service) SubmitBatchTransfer(ctx context.Context, batchID *uuid.U
 	payload := bitflyer.WithdrawToDepositIDBulkPayload{
 		Withdrawals: withdraws,
 	}
-	// upload
-	_, err = service.bfClient.UploadBulkPayout(ctx, payload)
+
+	withdrawToDepositIDBulkResponse, err := service.bfClient.UploadBulkPayout(ctx, payload)
 	if err != nil {
-		// if this was a bitflyer error and the error is due to a 401 response, refresh the token
-		var bfe *clients.BitflyerError
-		if errors.As(err, &bfe) {
-			if bfe.HTTPStatusCode == http.StatusUnauthorized {
-				// try to refresh the token and go again
+		var bitflyerError *clients.BitflyerError
+
+		switch {
+		case errors.As(err, &bitflyerError):
+
+			// if this was a bitflyer 401 response refresh the token and retry upload otherwise return bitflyer error
+			if bitflyerError.HTTPStatusCode == http.StatusUnauthorized {
 				logger.Warn().Msg("attempting to refresh the bf token")
 				_, err = service.bfClient.RefreshToken(ctx, bitflyer.TokenPayloadFromCtx(ctx))
 				if err != nil {
 					return fmt.Errorf("failed to get token from bf: %w", err)
 				}
-				// redo the request after token refresh
-				_, err := service.bfClient.UploadBulkPayout(ctx, payload)
+				withdrawToDepositIDBulkResponse, err = service.bfClient.UploadBulkPayout(ctx, payload)
 				if err != nil {
 					return fmt.Errorf("failed to transfer funds: %w", err)
 				}
+			} else {
+				return bitflyerError
 			}
 
-			for _, v := range bfe.ErrorIDs {
-				// non-retry errors, report to sentry
-				if v == "NO_INV" {
-					logger.Error().Err(bfe).Msg("no bitflyer inventory")
-					sentry.CaptureException(bfe)
```

Comment:
can we keep the logging to sentry on no_inv?

Reply:
It should get logged when it returns to the [RunNextBatchPaymentsJob](https://github.com/brave-intl/bat-go/blob/c3c1b621ece0b8cc5ca5981d1667c914fb169e1d/promotion/datastore.go#L1008)